src/tron/tronEngine.js

```diff
@@ -0,0 +1,641 @@
+            }
+            return { server, result }
+          }
+        )
+        // Randomize array
+        funcs = shuffleArray(funcs)
+        out = await asyncWaterfall(funcs)
+        break
+
+      case 'trx_getBalance':
+      case 'trx_getTransactions':
+        funcs = this.currencyInfo.defaultSettings.otherSettings.tronApiServers.map(
+          server => async () => {
+            const result = await this.fetchGet(server + params[0])
+            if (typeof result !== 'object') {
+              const msg = `Invalid return value ${func} in ${server}`
+              this.log(msg)
+              throw new Error(msg)
+            }
+            return { server, result }
+          }
+        )
+        // Randomize array
+        funcs = shuffleArray(funcs)
+        out = await asyncWaterfall(funcs)
+        break
+    }
+    this.log(`TRX multicastServers ${func} ${out.server} won`)
+
+    return out.result
+  }
+
+  // // ****************************************************************************
+  // // Public methods
+  // // ****************************************************************************
+
+  async startEngine() {
+    this.engineOn = true
+    this.addToLoop('checkBlockchainInnerLoop', BLOCKCHAIN_POLL_MILLISECONDS)
+    this.addToLoop('checkAccountInnerLoop', ACCOUNT_POLL_MILLISECONDS)
+    this.addToLoop('checkUpdateNetworkFees', NETWORKFEES_POLL_MILLISECONDS)
+    this.addToLoop('checkTransactionsInnerLoop', TRANSACTION_POLL_MILLISECONDS)
+    super.startEngine()
+  }
+
+  async resyncBlockchain() {
+    await this.killEngine()
+    await this.clearBlockchainCache()
+    await this.startEngine()
+  }
+
+  async makeSpend(edgeSpendInfoIn: EdgeSpendInfo) {
+    try {
+      const { edgeSpendInfo, currencyCode } = super.makeSpend(edgeSpendInfoIn)
+      const spendTarget = edgeSpendInfo.spendTargets[0]
+      const publicAddress = spendTarget.publicAddress
+      const nativeAmount = edgeSpendInfo.spendTargets[0].nativeAmount
+      let networkFee = await this.calcTxFee(publicAddress)
+      networkFee = bns.mul(networkFee, NATIVE_UNIT_MULTIPLIER)
+
+      const balanceTrx = this.walletLocalData.totalBalances[
+        this.currencyInfo.currencyCode
+      ]
+
+      if (bns.gt(nativeAmount, balanceTrx)) {
+        throw new InsufficientFundsError()
+      }
+
+      // Tron can only have one output
+      if (edgeSpendInfo.spendTargets.length !== 1) {
+        throw new Error('Error: only one output allowed')
+      }
+
+      const data =
+        spendTarget.otherParams != null ? spendTarget.otherParams.data : void 0 // eslint-disable-line no-void
+
+      let otherParams: TronTxOtherParams = {
+        visible: false,
+        txID: '',
+        raw_data: {},
+        raw_data_hex: ''
+      }
+      // if TRON
+      if (currencyCode === PRIMARY_CURRENCY) {
+        // builds tx, but does NOT send it
+        const trxParams: TronTxOtherParams = await tronWeb.transactionBuilder.sendTrx(
```

Comment:
It appears that the calculated networkFee isn't being included in the transactionBuilder. Where is the transactionBuilder getting the fee? There's a concern where the actual fee used will not match what we show the user.